src/helpers.ts

```diff
@@ -0,0 +1,61 @@
+import {NextFunction, Request, Response} from "express";
+import fetch from "node-fetch";
+
+class ErrorWithRedirect extends Error {
+  redirectTo: string
+
+  constructor(redirectTo: string) {
+    super(`There was an error that is solvable by redirecting to ${redirectTo}.`);
+    this.redirectTo = redirectTo
+  }
+}
+
+export const withResource = ({
```

Comment:
I think this is a pretty good idea (writing a wrapper around the fetching part). I've just checked out the SDK we generate for typescript, and I think it's pretty good and easy to use:

```
import kratos from '@oryd/kratos-client'
import {HttpError} from "@oryd/kratos-client/api/apis";

const api = new kratos.AdminApi('http://path/to/kratos')
api.getSelfServiceBrowserLoginRequest('asdf')
    .then(({response, body}) => {
    // body.methods.password.config.action
})
    .catch((err) => {
        if (err instanceof HttpError) {
            // err.statusCode == 404
        }
    })
```

What do you think about pausing this PR, moving to the SDK for auth endpoints and once that is merged, use the same methodology for the profile management?

Benefits would include:

- Future updates to paths just need a bump in the SDK
- Payloads are already pre-defined (we can delete the types.ts file)
- Easier to use and understand for developers

Reply:
Good idea